src/industry_cmd.cpp

```diff
@@ -1395,6 +1395,7 @@ static CommandCost CheckIfIndustryTilesAreFree(TileIndex tile, const IndustryTil
 			}
 		} else {
 			CommandCost ret = EnsureNoVehicleOnGround(cur_tile);
+			if (ret.Succeeded()) ret = EnsureNoShipFromDiagDirs(cur_tile);
```

Comment:
This one is different from the others, everywhere else just uses `AddCost` to merge error states.

Reply:
I'm not sure what you mean here. It looks fine, for me. Doesn't look like it need to use `AddCost`.